src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -0,0 +1,200 @@
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Comment } from '../../types/Comment';
+
+interface Props {
+  error: boolean;
+  loading: boolean;
+  onAdd: (comment: Omit<Comment, 'postId'>) => void;
+}
+
+export const NewCommentForm: React.FC<Props> = ({ error, loading, onAdd }) => {
+  const [name, setName] = useState('');
+  const [hasNameError, setHasNameError] = useState(false);
+
+  const [email, setEmail] = useState('');
+  const [hasEmailError, setHasEmailError] = useState(false);
+
+  const [body, setBody] = useState('');
+  const [hasBodyError, setHasBodyError] = useState(false);
+
+  const reset = () => {
+    setName('');
+    setEmail('');
+    setBody('');
+
+    setHasNameError(false);
+    setHasEmailError(false);
+    setHasBodyError(false);
+  };
+
+  const handleOnChangeName = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setHasNameError(false);
+
+    setName(event.target.value);
+  };
+
+  const handleOnChangeEmail = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setHasEmailError(false);
+
+    setEmail(event.target.value);
+  };
+
+  const handleOnChangeBody = (
+    event: React.ChangeEvent<HTMLTextAreaElement>,
+  ) => {
+    setHasBodyError(false);
+
+    setBody(event.target.value);
+  };
+
+  const handleOnSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setHasNameError(!name);
+    setHasEmailError(!email);
+    setHasBodyError(!body);
+
+    if (!name || !email || !body) {
+      return;
+    }
+
+    onAdd({
+      name,
+      email,
+      body,
+      id: 0,
```

Comment:
The `id` field is hardcoded to `0` when calling `onAdd`. Ensure that this is intentional. If the `id` is not needed, consider removing it from the object. If it is needed, ensure it is assigned a unique value.